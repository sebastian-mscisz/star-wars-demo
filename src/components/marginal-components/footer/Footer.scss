@import 'src/variables';

.sw-footer {
    display: flex;
    align-items: center;
    justify-content: center;
    height: $footer-height;
    color: $base-color;
    width: 100%;
    border-top: 2px solid $base-color;
    background-color: #0a0b0b;
    box-sizing: border-box;
    margin-top: auto;

    &__text {
        text-align: center;
        margin: 0px;
    }
}
