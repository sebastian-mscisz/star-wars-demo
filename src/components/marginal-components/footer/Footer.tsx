import './Footer.scss';

export const Footer: React.FC = () => {
    const getCurrentYear = new Date().getFullYear();

    return (
        <footer className="sw-footer">
            <p className="sw-footer__text">Â©{getCurrentYear} Star Wars Demopedia</p>
        </footer>
    );
};
