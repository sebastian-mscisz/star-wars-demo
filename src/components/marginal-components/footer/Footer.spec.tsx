import { BrowserRouter } from 'react-router-dom';
import { Footer } from './Footer';
import { render } from '@testing-library/react';

describe('Footer', () => {
    it('should render', () => {
        const { asFragment } = render(<Footer />, { wrapper: BrowserRouter });
        expect(asFragment()).toMatchSnapshot();
    });
});
