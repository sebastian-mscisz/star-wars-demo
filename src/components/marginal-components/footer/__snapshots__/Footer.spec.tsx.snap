// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render 1`] = `
<DocumentFragment>
  <footer
    class="sw-footer"
  >
    <p
      class="sw-footer__text"
    >
      Â©2023 Star Wars Demopedia
    </p>
  </footer>
</DocumentFragment>
`;
